### PR TITLE
Button

### PR DESCRIPTION
**Pull request by quentinv72**
Refactored button component. Added the hover to the component.

### CODE REVIEW

File: src/components/button.js [hunk 1]
```diff
@@ -2,11 +2,21 @@ AFRAME.registerComponent("button", {
   schema: {
     value: { type: "string", default: "button" },
     height: { type: "number", default: 0.5 },
-    width: { type: "number", default: 4 },
+    widthText: { type: "number", default: 4 },
```
Comment (amirhmk) [inline src/components/button.js:5]: What do you think about `textWidth`? 😈

File: src/components/button.js [hunk 2]
```diff
@@ -15,15 +25,24 @@ AFRAME.registerComponent("button", {
...
 
     el.setAttribute("text", {
       value: value,
       align: "center",
-      width: width,
+      width: widthText,
+    });
+    el.setAttribute("class", "clickable");
+    el.addEventListener("mouseenter", function () {
+      el.setAttribute("geometry", { width: 1.6, height: 0.6 });
```
Comment (amirhmk) [inline src/components/button.js:42]: Would these hard-coded values work with any other `borderRadius`?

File: src/components/button.js (hunk 1 above)
Comment (quentinv72) [reply src/components/button.js:5]: Hahah it does sound a bit better

File: src/components/button.js (hunk 2 above)
Comment (quentinv72) [reply src/components/button.js:42]: They should but I'm gonna change that so that they aren't hard coded.

**Comment by quentinv72** [comment]
I think we can just set gaze to false in the cursor component. It does not seem like we will need that feature. Or maybe we could have it activate only if a person is using their phone or a remoteless vr?

**Comment by amirhmk** [comment]
> I think we can just set gaze to false in the cursor component. It does not seem like we will need that feature. Or maybe we could have it activate only if a person is using their phone or a remoteless vr?

I would say we don't need it, especially given that it's mostly useful for mobile phones?

**Pull request closed without merging**